### PR TITLE
Added --allow_origin option

### PR DESCRIPTION
**Pull request by zischwartz**
Per [the discussion here](https://github.com/odewahn/tmpnb/commit/3edc90a5e741144d95ba4325ec8e425e9f4c5a72#commitcomment-9231242) and #115, I added this as a command line option. 

This also, unrelatedly, upgrades pip, because I was having the same problems as noted in #130.

### CODE REVIEW

**Comment by Carreau** [comment]
Seem legit. I'll let @rgbkrk give final word.

File: README.md [hunk 1]
```diff
@@ -53,7 +53,9 @@ docker run --net=host -d -e CONFIGPROXY_AUTH_TOKEN=$TOKEN \
 Usage: orchestrate.py [OPTIONS]
 
 Options:
-
+  --allow_origin                   Set the Access-Control-Allow-Origin header.
```
Comment (rgbkrk) [inline README.md]: This works with `--allow-origin` too right? I'd rather see the `-` form for what users see.

**Comment by rgbkrk** [comment]
Thanks for this.

Since we need it across several, perhaps we should create a BaseHandler that has the `allow_origin` option so we don't repeat ourselves.

File: README.md (hunk 1 above)
Comment (zischwartz) [inline README.md]: I'm not sure, I haven't used Tornado a ton tbh. I was just trying to follow the convention used by the other options (e.g. `--pool_size `). 

And I named it that to keep it consistent with what jupyter/ipython uses (`--NotebookApp.allow_origin`)

Comment (rgbkrk) [inline README.md]: Well we can blame me then. :wink:

**Comment by rgbkrk** [comment]
Darn it, I just merged the other one and now there are conflicts. Are you comfortable with rebasing or would you like me to merge it by hand?

**Comment by zischwartz** [comment]
I just pulled and merged master into my fork, so it should merge cleanly now.

**Comment by rgbkrk** [comment]
Thank you @zischwartz!

**Pull request merged**